include/threads.h

```diff
@@ -0,0 +1,102 @@
+/*
```

Comment:
inldude/threads.h? hmmm....

Reply:
i tried moving it to `libc/common/include` but then picolibc gave me an error.

Reply:
IMO this needs some discussion and broader review, putting this at this level is just a source for problems and confusion.